sql/sql_utils.cc

```diff
@@ -38,6 +43,65 @@ std::string escapeSql(const char * str)
     return escapeSql(std::string(str));
 }
 
+bool matchSqlFilter(const Utf8String& valueString, const Utf8String& filterString)
+{
+    if (filterString.empty() || valueString.empty())
+        return false;
+
+    Utf8String regExFilter;
+
+    for (const auto& filterChar : filterString) {
```

Comment:
See below, [] are not part of like. I'll check for the $ and ^, not sure which standard boost is running.